sdk/core/Azure.Core/src/DynamicData/DynamicDataOptions.cs

```diff
@@ -38,6 +38,7 @@ internal static JsonSerializerOptions ToSerializerOptions(DynamicDataOptions opt
                     new DynamicData.DynamicTimeSpanConverter(),
                     new DynamicData.DynamicDateTimeConverter(options.DateTimeFormat),
                     new DynamicData.DynamicDateTimeOffsetConverter(options.DateTimeFormat),
+                    new ModelJsonConverter(),
```

Comment:
Is it easy/possible to only add this when we're looking at IModelSerializable types?  There is a perf overhead for each added converter.